server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/utils/iterator/SortedNumericStarTreeValuesIterator.java

```diff
@@ -29,4 +29,12 @@ public SortedNumericStarTreeValuesIterator(DocIdSetIterator docIdSetIterator) {
     public long nextValue() throws IOException {
         return ((SortedNumericDocValues) docIdSetIterator).nextValue();
     }
+
+    public int valuesCount() throws IOException {
```

Comment:
nit : can we have 1-1 mapping , lets have the same name `docValueCount`

Reply:
These are "star tree values", though, right?

Reply:
I wanted to avoid `docValue` within the method name. 

We also didn't had 1-1 mapping in the parent class as well.

```
    public int entryId() {
        return docIdSetIterator.docID();
    }

    public int nextEntry() throws IOException {
        return docIdSetIterator.nextDoc();
    }
```

`valuesCount()` looked kind of okay to me, but okay to rename otherwise.

Reply:
i mean we can have 'entryValueCount' going by that convention - the thing is we have getValueCount and getDocValueCount for sortedSetDocValues - so just `valuesCount` will have issues.

```
  /**
   * Retrieves the number of unique ords for the current document. This must always be greater than
   * zero. It is illegal to call this method after {@link #advanceExact(int)} returned {@code
   * false}.
   */
  public abstract int docValueCount();

  /**
   * Returns the number of unique values.
   *
   * @return number of unique values in this SortedDocValues. This is also equivalent to one plus
   *     the maximum ordinal.
   */
  public abstract long getValueCount();
```

Reply:
Cool, renamed to `entryValueCount`. Resolving this.